packages/qr_api/lib/src/api/qr_api.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:dio/dio.dart';
+import 'package:qr_api/src/api/api_result.dart';
+import 'package:qr_models/qr_models.dart';
+
+/// {@template qr_api}
+/// This works as a remote data source for the QrSeed requests
+/// {@endtemplate}
+class QrApi {
+  /// Default construvctor with a [Dio] instance
+  QrApi(this._dio);
+
+  final Dio _dio;
+
+  /// This return a [ApiResult] state with the model [QRSeed] as value
+  Future<ApiResult<QRSeed>> fetchQrApi() async {
+    try {
+      final response =
+          await _dio.get<Map<String, dynamic>>('/default/random-qr-seed_seed');
+      final seed = QRSeed.fromJson(response.data!);
+      return ApiResult.success(seed);
```

Comment:
chore: we should be checking the data validity where possible and returning a failure. Exceptions carry a ton of baggage with them that severely hurts performance of applications and we shouldn't rely on them when we can inspect the fields and return errors.